packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel_cloud_firestore.dart

```diff
@@ -0,0 +1,356 @@
+    // Delegate handling to the implementation
+    return await transactionHandler(transactionId);
+  }
+
+  // Document Reference
+  @override
+  Future<void> setDocumentReferenceData(String app, {
+    @required String path,
+    // TODO: Type SetOptions: https://firebase.google.com/docs/reference/js/firebase.firestore.SetOptions.html
+    Map<String, dynamic> options,
+    Map<String, dynamic> data = const <String, dynamic>{},
+  }) => channel.invokeMethod<void>(
+      'DocumentReference#setData',
+      <String, dynamic>{
+        'app': app,
+        'path': path,
+        'data': data,
+        'options': options,
+      },
+    );
+
+  @override
+  Future<void> updateDocumentReferenceData(String app, {
+    @required String path,
+    Map<String, dynamic> data = const <String, dynamic>{},
+  }) => channel.invokeMethod<void>(
+      'DocumentReference#updateData',
+      <String, dynamic>{
+        'app': app,
+        'path': path,
+        'data': data,
+      },
+    );
+
+  // TODO: Type this return
+  @override
+  Future<Map<String, dynamic>> getDocumentReference(String app, {
+    @required String path,
+    @required String source,
+  }) => channel.invokeMapMethod<String, dynamic>(
+      'DocumentReference#get',
+      <String, dynamic>{
+        'app': app,
+        'path': path,
+        'source': source,
+      },
+    );
+
+  @override
+  Future<void> deleteDocumentReference(String app, {
+    @required String path,
+  }) => channel.invokeMethod<void>(
+      'DocumentReference#delete',
+      <String, dynamic>{'app': app, 'path': path},
+    );
+
+  // TODO: Port to stream
+  @override
+  Future<int> addDocumentReferenceSnapshotListener(String app, {
+    @required String path,
+    bool includeMetadataChanges,
+  }) => channel.invokeMethod<int>(
+          'DocumentReference#addSnapshotListener',
+          <String, dynamic>{
+            'app': app,
+            'path': path,
+            'includeMetadataChanges': includeMetadataChanges,
+          },
+        );
+
+  // Query
+  Future<int> _addQuerySnapshotListener(String app, {
+    @required String path,
+    bool isCollectionGroup,
+    Map<String, dynamic> parameters,
+    bool includeMetadataChanges,
+  }) => channel.invokeMethod<int>(
+          'Query#addSnapshotListener',
+          <String, dynamic>{
+            'app': app,
+            'path': path,
+            'isCollectionGroup': isCollectionGroup,
+            'parameters': parameters,
+            'includeMetadataChanges': includeMetadataChanges,
+          },
+        );
+
+  static final Map<int, StreamController<int>> _queryObservers =
+      <int, StreamController<int>>{};
+
+  @override
+  Stream<dynamic> snapshots(String app, {
+    @required String path,
+    bool isCollectionGroup,
+    Map<String, dynamic> parameters,
+    bool includeMetadataChanges,
+  }) {
+    // Create a stream of query snapshot handles, as they happen
+        assert(includeMetadataChanges != null);
```

Comment:
```suggestion
    assert(includeMetadataChanges != null);
```

Reply:
I ran dartfmt afterwards, I'll try to get everything better formatted by GM